### PR TITLE
fix(auth): Invalid unblock code after swapping primary email

### PR DESCRIPTION
**Pull request by dschom**
## Because

- We found an edge case where the unblock code was consistently considered invalid

## This pull request

- Fixes occasional race condition in consume unblock code stored procedure.
- Fixes a small bug in the test framework where the most recently emailed unblock code wasn't being used.

## Issue that this pull request solves

Closes: #12457

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Review by vbudhram — APPROVED** [review]
@dschom Thanks for the PR, this LGTM. Please update the deploy doc to let ops know to run the migration.

It looks like CI is having an outage so these might be errors related to that.

File: packages/fxa-content-server/tests/functional/sign_in_blocked.js [hunk 1]
```diff
@@ -171,12 +171,13 @@ registerSuite('signin blocked', {
           )
 
           .then(fillOutSignInUnblock(email, 0))
-          // the first code is no longer valid, must use the 2nd.
-          .then(visibleByQSA(selectors.SIGNIN_UNBLOCK.ERROR))
-          .then(testErrorTextInclude('invalid'))
 
+          // If these checks are in place, the edge case where we have swapped primary accounts fails.
```
Comment (dschom) [inline packages/fxa-content-server/tests/functional/sign_in_blocked.js:175]: Not sure about this. As we discussed it sounds like burning the unblock code even on a bad login attempt is intentional. So this probably isn't acceptable and another approach should be taken.

**Comment by vbudhram** [comment]
closing in favor of https://github.com/mozilla/fxa/pull/12757
